Fix issues when a float is used instead of a number

nightwolfz opened the pull request:
Some tests for MQTT.js use a float for the `keepalive` argument for mqtt.connect()

This fixes pending connections.

```
1) Websocket Client pinging should defer the next ping when sending a control packet:
     Error: timeout of 5000ms exceeded. Ensure the done() callback is being called in this test.
```

mcollina commented:
May I ask you to add a unit test? And also a comment in there?
Il giorno mar 19 gen 2016 alle 19:31 Ryan M. <notifications@github.com> ha
scritto:

> Some tests for MQTT.js use a float for the `keepalive
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/mqttjs/mqtt-packet/pull/10
> Commit Summary
>
>    - Fix issues when a float is used instead of a number
>
> File Changes
>
>    - *M* writeToStream.js
>    <https://github.com/mqttjs/mqtt-packet/pull/10/files#diff-0> (2)
>
> Patch Links:
>
>    - https://github.com/mqttjs/mqtt-packet/pull/10.patch
>    - https://github.com/mqttjs/mqtt-packet/pull/10.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mqttjs/mqtt-packet/pull/10>.
>

nightwolfz commented:
Apologies, added a small test case and changed the code to prevent unexpected behavior (throw an error instead of rounding off floats to integers like in the previous commit). 

The performance seems unaffected.

mcollina commented:
Thanks!!
